extensions/spring-xd-extension-header-enricher/src/test/java/org/springframework/xd/header/enricher/HeaderEnricherIntegrationTests.java

```diff
@@ -78,6 +83,25 @@ public void testSimpleExpression() {
 	}
 
 	@Test
+	public void testOverwriteHeaders() {
+		String streamName = "testOverwrite";
+
+		String headers = "{\"bar\":\"payload.substring(1)\"}";
+
+		String processingChainUnderTest = String.format("%s --headers=%s --overwrite=true", moduleName, headers);
```

Comment:
Not sure what's going on but if I change this to `false`, this test fails as expected, but so does `testSimpleExpression` with an NPE at line 77.